packages/inventory/src/services/inventory.ts

```diff
@@ -14,33 +14,66 @@ import {
   InventoryItemDTO,
   ReservationItemDTO,
   InventoryLevelDTO,
+  TransactionBaseService,
+  ConfigurableModuleDeclaration,
+  MODULE_RESOURCE_TYPE,
 } from "@medusajs/medusa"
 
 import {
   InventoryItemService,
   ReservationItemService,
   InventoryLevelService,
 } from "./"
+import { EntityManager } from "typeorm"
 
 type InjectedDependencies = {
+  manager: EntityManager
   eventBusService: IEventBusService
 }
 
-export default class InventoryService implements IInventoryService {
+export default class InventoryService
+  extends TransactionBaseService
+  implements IInventoryService
+{
   protected readonly eventBusService_: IEventBusService
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
   protected readonly inventoryItemService_: InventoryItemService
   protected readonly reservationItemService_: ReservationItemService
   protected readonly inventoryLevelService_: InventoryLevelService
 
-  constructor({ eventBusService }: InjectedDependencies) {
+  constructor(
+    { eventBusService, manager }: InjectedDependencies,
+    options?: unknown,
+    moduleDeclaration?: ConfigurableModuleDeclaration
+  ) {
+    super(arguments[0])
+
     this.eventBusService_ = eventBusService
 
-    this.inventoryItemService_ = new InventoryItemService({ eventBusService })
+    if (
+      moduleDeclaration?.resources === MODULE_RESOURCE_TYPE.SHARED &&
+      manager
+    ) {
+      this.manager_ = manager
+    } else {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_ARGUMENT,
+        "At the moment this module can only be used with shared resources"
+      )
+    }
+
+    this.inventoryItemService_ = new InventoryItemService({
```

Comment:
thought: It is a shame that we are losing the injection for that service. It is not a good practice to instantiate a class from the constructor as it hides those and creates unnecessary complexity to test. I would have thought that the module would have at least its own container to register all those services. Would it make sense? something like 
`createContainer({ something }, coreContainer)` but since all resources are shared should we export all the services for the core to register them? what is you opinion?

Reply:
The idea here is that the `InventoryItemService`, etc. are meant to be hidden from the global container and only accessible through one API (namely, the `IInventoryServce`). A separate container for the module could make sense - although I am not sure how we would then handle the InventoryService which would need injections from both the "core container" and the module container.

Reply:
we can do something like that:
The loader already accept the container from the core in the options:
- Create a local container that take the core container as the second argument
- register the different services from the module
- resolve the `InventoryItemService`
- register it

```ts
const container = createContainer({}, coreContainer)
container.register("serviceA", asFunction((cradle) => new ServiceA(cradle)))
container.register("serviceB", asFunction((cradle) => new ServiceB(cradle))) // this service is injected into A

const serviceA = container.resolve("serviceA")

// either that
coreContainer.register("serviceA", asValue(serviceA))
// or another approach

```

what do you think? I did not test it, it is just pseudo code

Reply:
It is very similar to what I do in the tests

Reply:
I don't think the core should import more than what is expected. These 2 modules will have their own container.
They don't have it yet because the tests are not in place. Next steps will be integrate the modules in our integration-tests, check the best way to add integration tests in these modules as well + the unit tests

Reply:
I am fine with holding off with module containers for now.

Reply:
I am fine too for now :D